ouroboros-network/src/Ouroboros/Network/ChainFragment.hs

```diff
@@ -111,6 +114,17 @@ import           Ouroboros.Network.Point (WithOrigin (At))
 newtype ChainFragment block = ChainFragment (FingerTree BlockMeasure block)
   deriving (Show, Eq)
 
+-- | The 'FingerTree' might have internal thunks
+--
+-- TODO: Ideally we should just give an instance for 'FingerTree' directly
+-- (but without introducing an orphan).
+instance NoUnexpectedThunks block
+      => NoUnexpectedThunks (ChainFragment block) where
+  showTypeOf _ = "ChainFragment"
+  whnfNoUnexpectedThunks ctxt (ChainFragment ft) =
+      allNoUnexpectedThunks $
```

Comment:
Could have used `noUnexpectedThunksInValues` here :man_shrugging:

Reply:
I suggest we use it